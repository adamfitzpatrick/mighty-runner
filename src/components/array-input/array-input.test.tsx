import * as React from 'react'
import { cleanup, render, RenderResult, fireEvent } from '@testing-library/react'
import ArrayInput from '.'

describe('ArrayInput component', () => {
  let sut: RenderResult
  let arr: string[]
  let change: jest.Mock<string[]>
  let blur: jest.Mock

  beforeEach(() => {
    arr = [ 'a', 'b', 'c' ]
    change = jest.fn()
    blur = jest.fn()
    sut = render(<ArrayInput label='label' arr={ arr } onChange={ change } onBlur={ blur } />)
  })

  test('should render correctly', () => {
    expect(sut.container.innerHTML).toMatchSnapshot()
  })

  test('should call the change handler with a new array for all changes', () => {
    const secondInput = sut.getByTestId('1.array.input')
    fireEvent.change(secondInput, { target: { value: 'change' } })
    const changeCallArg = change.mock.calls[0][0]
    expect(changeCallArg[1]).toBe('change')
    expect(changeCallArg).not.toBe(arr)
  })

  test('should call the blur handler', () => {
    const secondInput = sut.getByTestId('1.array.input')
    fireEvent.blur(secondInput)
    expect(blur).toHaveBeenCalled()
  })
})
