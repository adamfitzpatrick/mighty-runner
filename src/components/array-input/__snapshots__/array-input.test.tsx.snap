// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrayInput component should render correctly 1`] = `"<label>label<input data-testid=\\"0.array.input\\" type=\\"text\\" value=\\"a\\"></label><label><span class=\\"additionalLabel\\">label</span><input data-testid=\\"1.array.input\\" type=\\"text\\" value=\\"b\\"></label><label><span class=\\"additionalLabel\\">label</span><input data-testid=\\"2.array.input\\" type=\\"text\\" value=\\"c\\"></label>"`;
